packages/stories/src/gitgraph-js/6-custom-renders.stories.tsx

```diff
@@ -300,4 +306,235 @@ storiesOf("gitgraph-js/6. Custom renders", module)
+              bgColor: 'orange',
+              strokeColor: 'orange',
+              borderRadius: 0,
+              pointerWidth: 0,
+            };
+            gitgraph.tag({
+              name: 'last release',
+              style: customTagStyle,
+            });
+            gitgraph
+              .branch('feat1')
+              .commit("Message appears below the tags")
+              .tag({ name: 'something cool', style: customTagStyle });
+          }}
+        </GraphContainer>
+    )
+  })
+  .add("custom renderer - background rect", () => {
+    // Providing a custom Renderer class can enable limitless customization.
+    // However -
+    // NOTE: using a custom Renderer can be considered advanced and experimental.
+    //   1. You will have to read the code to figure out which methods to override.
+    //   2. The internal API (ie. the Renderer methods and variables) is *internal* and
+    //      may change in the future.
+    //
+    // In this example we add a background rectangle to the commit which will change
+    // fill-color on hover and if selected.
+    //
+    // HOVER over the commits, and CLICK to see the background change
+
+    let selectedCommit: Commit|null = null;
+    let renderer: any;
+    // CustomRenderer class - note the vanilla JS syntax as it appears in storybook
+    // is "ugly" but modern browsers support the `class` keyword which provides
+    // prettier syntax. The source code for this story uses the ES6 class, which gets
+    // transpiled to the prototype-based syntax you see in storybook component explorer.
+    class BackgroundRectRenderer extends Renderer {
+        createCommitGroup(commit: Commit, {x, y}: Coordinate): SVGGElement {
+          const result = super.createCommitGroup(commit, {x,y})
+          result.setAttribute("class", commit === selectedCommit ? "commit-row selected" : "commit-row")
+          result.addEventListener("click", function() {
+            selectedCommit = commit;
+            // Using the `rerender` method will render again
+            // using the same data as before - on modern browsers
+            // it will keep the vertical scroll location
+            renderer.rerender()
+          })
+          return result;
+        }
+        getCommitGroupChildren(commit: Commit, {x, y}: Coordinate): Array<SVGElement | null> {
+          // we have to provide explicit dimensions for the background rect
+          // (unfortunately SVG does not support having a rect as a expanding container)
+          let height = 0;
+          // find the minimum height (Y-distance to parent)
+          this.getParentCommits(commit).forEach(parent => {
+            if (parent) {
+              const parentY = this.getWithCommitOffset(parent).y;
+              height = Math.max(height, parentY - y);
+            }
+          })
+          const rect = createRect({
+            width: this.svg.clientWidth-20,
+            height: height,
+            translate: {x:-6, y:-6}
```

Comment:
Any rationale behind these numbers (20 and 6)?

Reply:
I tweaked the numbers until it looked good... I didn't spend time to find deep reasoning behind them.

I wanted the rect to fill the entire width, but I'm not sure why I had to take down 20px.

There is some tricky bit here - 
if the rect width is bigger than the svg width, then the svg grows to accommodate it. Then on second rerender (ie. when clicked) the rectangle is even bigger (because it is based on the svg width) and so repeat clicks make the svg/rects grow more and more.
So its quite important for the `rect` to be no more than the svg.  I found that -20 is what I have to use to avoid the repeat-click-growing-wider problem.  I'm not sure where the 20px come from, it feels like a 10px padding, but clientWidth should include the padding in it.

Reply:
I see, I can relate to that kind of weird thing 😆 

Maybe just drop a comment to explain that these are the values you found work best to have the rect fill the entire width, but we're not sure what they correspond to yet.

Reply:
oh, I realized the reason  - it is the 10px padding!  when the rect is being sized to the svg width, then the CSS rendering adds padding on top of the rect width, making the SVG larger when it updates the DOM!  That is why the padding needs to be subtracted from the current width.  I'll add a comment explaining the numbers.